static/app/views/dashboardsV2/widgetBuilder/buildSteps/filterResultsStep/index.tsx

```diff
@@ -73,17 +73,20 @@ export function FilterResultsStep({
     };
   }, []);
 
-  const handleBlur = useCallback((queryIndex: number) => {
-    return (field: string) => {
-      if (!blurTimeoutRef.current) {
-        const newQuery: WidgetQuery = {
-          ...queries[queryIndex],
-          conditions: field,
-        };
-        onQueryChange(queryIndex, newQuery);
-      }
-    };
-  }, []);
+  const handleBlur = useCallback(
+    (queryIndex: number) => {
+      return (field: string) => {
+        if (!blurTimeoutRef.current) {
+          const newQuery: WidgetQuery = {
+            ...queries[queryIndex],
+            conditions: field,
+          };
+          onQueryChange(queryIndex, newQuery);
+        }
+      };
+    },
+    [queries]
```

Comment:
oh that was the queries... funny, I  would have fixed this in the PR https://github.com/getsentry/sentry/pull/33241

Reply:
we have to do the same with `handleChange` 😉